src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -170,6 +174,11 @@ public int Seek(int char0)
                 }
                 while (block.End != index)
                 {
+                    if (_processed > _maxHeaderLength)
```

Comment:
We could put these limits directly on ServiceContext like we do for ConnectionFilter. It might be best to put the entire IKestrelServerInformation object in ServiceContext.